Fix:

error: type specifier missing, defaults to 'int'; ISO C99 and later do not support implicit int [-Wimplicit-int]
error: call to undeclared library function 'exit' with type 'void (int) __attribute__((noreturn))'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
--- configure.orig	2003-10-04 20:55:22.000000000 -0500
+++ configure	2024-10-05 01:51:16.000000000 -0500
@@ -655,7 +655,7 @@
 #line 656 "configure"
 #include "confdefs.h"
 
-main(){return(0);}
+int main(void){return(0);}
 EOF
 if { (eval echo configure:661: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext}; then
   ac_cv_prog_cc_works=yes
@@ -1396,6 +1396,7 @@
 #line 1397 "configure"
 #include "confdefs.h"
 #include <ctype.h>
+#include <stdlib.h>
 #define ISLOWER(c) ('a' <= (c) && (c) <= 'z')
 #define TOUPPER(c) (ISLOWER(c) ? 'A' + ((c) - 'a') : (c))
 #define XOR(e, f) (((e) && !(f)) || (!(e) && (f)))
@@ -1627,7 +1628,7 @@
 
 #include <stdio.h>
 #include <sys/types.h>
-main() { return (sizeof(off_t) == 4); }
+int main(void) { return (sizeof(off_t) == 4); }
 EOF
 if { (eval echo configure:1633: \"$ac_link\") 1>&5; (eval $ac_link) 2>&5; } && test -s conftest${ac_exeext} && (./conftest; exit) 2>/dev/null
 then
@@ -1663,8 +1664,8 @@
 #line 1664 "configure"
 #include "confdefs.h"
 
-int main() {
-} $ac_kw foo() {
+int main(void) {
+} $ac_kw int foo(void) {
 ; return 0; }
 EOF
 if { (eval echo configure:1671: \"$ac_compile\") 1>&5; (eval $ac_compile) 2>&5; }; then
